src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -1044,6 +1054,21 @@ void CouplingSchemeConfiguration::checkIfDataIsExchanged(
                 dataName);
 }
 
+int CouplingSchemeConfiguration::getWaveformUsedOrder(std::string dataName) const
+{
+  int maxUsedOrder = -1;
+  for (const precice::impl::PtrParticipant &participant : _participantConfig->getParticipants()) {
+    for (auto &dataContext : participant->readDataContexts()) {
+      if (dataContext.getDataName() == dataName) {
+        int usedOrder = dataContext.getInterpolationOrder();
+        PRECICE_ASSERT(usedOrder >= 0); // ensure that usedOrder was set
+        return usedOrder;
+      }
+    }
+  }
+  return maxUsedOrder;
+}
```

Comment:
I am a bit confused by the variable `maxUsedOrder` here.

Wouldn't it be "more correct" if we we pass the participant name as an argument here and return the order of that participant? In the end, we are only interested in the receiving participant, no?

Reply:
That's a leftover. You are 100% right. I will quickly fix it.